File: listable/src/main/java/com/apprikot/listable/utils/FontUtils.java
package com.apprikot.listable.utils;

import android.content.Context;
import android.graphics.Typeface;

import java.util.HashMap;
import java.util.Map;

public class FontUtils {
    public static final String CORBEL_BOLD = "Corbel Bold.ttf";
    public static final String ACCESSORIES = "Aladin-Regular.ttf";

    public enum CustomFont {
        CORBEL_BOLD(FontUtils.CORBEL_BOLD),
        ACCESSORIES(FontUtils.ACCESSORIES);

        private String mFontName;

        CustomFont(String fontName) {
            mFontName = fontName;
        }

        public String getFontName() {
            return mFontName;
        }
    }

    private static Map<CustomFont, Typeface> customFonts;

    public static Typeface getFont(Context context, CustomFont customFont) {
        if (customFonts == null) {
            customFonts = new HashMap<>();
        }

        if (!customFonts.containsKey(customFont) || customFonts.get(customFont) == null) {
            Typeface typeface = Typeface.createFromAsset(context.getAssets(), "fonts/" + customFont.getFontName());
            customFonts.put(customFont, typeface);
        }

        return customFonts.get(customFont);
    }
}

File: app/src/main/java/com/tacme/news/view/utils/FontUtils.kt
package com.tacme.news.view.utils

import android.content.Context
import android.graphics.Typeface
import com.tacme.news.networking.GlobalKeys.kAppLanguage
import com.tacme.news.networking.GlobalKeys.kDefaultLang
import com.tacme.news.utils.PreferencesHelper.Companion.getInstance
import java.util.*

object FontUtils {
    private var customFonts: MutableMap<CustomFont, Typeface>? =
        null

    @JvmStatic
    fun getFont(
        context: Context,
        customFont: CustomFont
    ): Typeface? {
        if (customFonts == null) {
            customFonts =
                HashMap()
        }
        val fontName = if (getInstance()[kAppLanguage, kDefaultLang]
            == "en"
        ) CustomFont.ENGLISH_MEDIUM.fontName else CustomFont.ARABIC_MEDIUM.fontName
        val typeface = Typeface.createFromAsset(context.assets, "fonts/$fontName")
        customFonts!![customFont] = typeface
        return customFonts!![customFont]
    }

    enum class CustomFont(
        val fontName: String
    ) {
        ENGLISH_MEDIUM("lato/Lato-Medium.ttf"),
        ARABIC_MEDIUM("bahe/Bahij_TheSansArabic-Plain.ttf");

    }
}